app/routes/agreement-submission.js

```diff
@@ -0,0 +1,27 @@
+const joi = require('joi')
+const { v4: uuidv4 } = require('uuid')
+const { sendAgreementSubmitMessage } = require('../messaging')
+
+module.exports = [{
+  method: 'POST',
+  path: '/agreement/submission',
+  options: {
+    validate: {
+      payload: joi.object({
+        agreementNumber: joi.string().required(),
+        sbi: joi.string().required(),
+        agreement: joi.object().required()
+      }),
+      failAction: async (request, h, error) => {
+        return h.response('Bad request').code(400).takeover()
+      }
+    },
+    handler: async (request, h) => {
+      const messageId = uuidv4()
+      await sendAgreementSubmitMessage(request.payload, messageId)
```

Comment:
Is the agreement already saved in the database?

If so it might be better to just supply an Id to this request and have this service pull the agreement out.  Rather than the web sending the full agreement?